### DISCUSSION TITLE
app_home/PS3_GAME - HEN

### DISCUSSION THREAD

**Discussion by ozapft** [aldostools/webMAN-MOD#664] [Q&A]
In setup.ps3 I can select the item /app_home.

But I don't see any changes regarding the app_home icon. 
With a normal HEN installation, app_home is not present in XMB regardless of this item is selected or not.

I´ve to edit the category_game.xml in /dev_flash/.. and insert 2 blocks:

\<Query
	class="type:x-xmb/folder-pixmap"
	key="seg_gamedebug"
	src="#seg_gamedebug"
	/>

and

\<View id="seg_gamedebug">
		\<Attributes>
			\<Table key="game_debug">
				\<Pair key="icon_rsc"><String>tex_album_icon</String></Pair>
				\<Pair key="title_rsc"><String>msg_tool_app_home_ps3_game</String></Pair>
				\<Pair key="child"><String>segment</String></Pair>
			\</Table>
		\</Attributes>
		\<Items>
			\<Query class="type:x-xcb/game-debug" key="game_debug" attr="game_debug"/>
		\</Items>
	\</View>

After these manual changes, * app_home/PS3_GAME is present in XMB and working.

However, the item /app_home in setup.ps3 still does nothing.
What is this selection supposed to do?

**Comment by aldostools** [comment]
The option /app_home has multiple functions:
- If checked /app_home is mapped to /dev_bdvd when an ISO is mounted
- If checked it also adds the app_home query to fb.xml (only if "seg_gamedebug" is not found in category_game.xml and "webMAN Games" is checked in /setup.ps3).
- If checked, ROMS are focused on /app_home (and launched if Auto-Play is checked)

**Comment by ozapft** [comment]
Thank you for your answer.

- If checked /app_home is mapped to /dev_bdvd when an ISO is mounted - **tested OK**
- If checked it also adds the app_home query to fb.xml (only if "seg_gamedebug" is not found in category_game.xml and "webMAN Games" is checked in /setup.ps3). **tested NOK**
If I use the original category_game.xml of HEN and /app_home is selected there is no query in fb-hen.xml. -> no /app_home icon in XMB
The orginal category_game.xml from HEN has no key seg_gamedebug. I checked this again.
There is no fb.xml. I think fb.xml is mapped to fb-hen.xml 

- If checked, ROMS are focused on /app_home (and launched if Auto-Play is checked) - **not tested**

**Comment by aldostools** [comment]
Thank you for your QA tests. I will review the issue about app_home icon on HEN when I finish a feature that I'm working on to reduce the memory usage on the full edition.

**Comment by ozapft** [comment]
Thank you for your answer!

Only one question about the function:

- If checked, ROMS are focused on /app_home (and launched if Auto-Play is checked) 

Do I need retroarch to test this function?

**Comment by aldostools** [comment]
Yesterday I tested the addition of app_home when using category_games.xml from HEN and webMAN added it to XMB (not to the file).

In regards to your question, you need RetroArch for a proper test (although it is not required) due the /app_home option in /setup.ps3 is a general setting that is not specific to RetroArch.

**Reply by ozapft** [reply]
Thank you for your efforts and this information.

This is strange. I have tested it several times. 
* app_home only appears in xmb when I edit category_game manually.
I have a normal and very clean installation. Only HFW 4.86, HEN 3.0.3, webMAN MOD and Irisman is installed.

**Comment by aldostools** [comment]
This was the way I tested it;
1- I replaced /dev_blind/vsh/resource/explore/xmb/category_games.xml with the one included in HEN
2- I verified that the /app_home setting was checked
3- I refreshed XMB to verify that ★ /app_home/PS3_GAME is not present (in CFW Evilnat it is named ★ PlayStation®3)
4- I rebooted the system
5- After reboot I had ★ PlayStation®3 (becaue I'm in CFW Evilnat)

**Reply by ozapft** [reply]
I did the same.
Today I´ve tested this steps with my old loud 2nd PS3 running CFW Evilnat.
I followed exactly the same steps, but the ★ /app_home/PS3_GAME (★ Playstation®3) icon didn't show up after the reboot. 
I´ve double checked all steps.

There must be a difference somewhere.

**Comment by aldostools** [comment]
I will recheck it again later this week.

**Reply by aldostools** [reply]
Btw I forgot to mention that I refreshed XML between  steps between 3 and 4.

 The reboot is not necessary. Reload XMB is enough after refresh XML.
